ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -447,8 +453,7 @@ private void checkCapacityAndUpdateReplicaStatusDelegate() {
         }
       }
       // During startup, we also need to reconcile the replica state from both ZK clusters.
-      if (!started && replicaStatusDelegates.size() > 1 && thresholdBytesLow < index.getLogUsedCapacity()
-          && index.getLogUsedCapacity() <= thresholdBytesHigh) {
+      if (!started && replicaStatusDelegates.size() > 1) {
```

Comment:
Did you find any case where the current logUsedCapacity < thresholdBytesLow(readwrite threshold) but the state is sealed and need to be unseal during startup? or any case logUsedCapacity > thresholdBytesHigh(readonly threshold) but state should change to unseal state during startup? Is this only used for zk migration since you want to keep the partition state same? If so, I would suggest you could use config to enable and disable this method once the migration finished. Since for each put and compaction we will tuning this state anyway.

Reply:
This code is executed only at startup. There is a node in ei4 atm that is not reconciling the states in zookeeper and I suspect its because of this threshold boolean. There is no easy way to test it unless we merge it. I am looking at some hacky ways of testing atm.

Reply:
Since it's for debugging purpose, can you not change the if condition to avoid logic change?
Can we only add debug message?

Reply:
i want the code to enter the if condition, i suspect its not entering this, hence i changed the logic

Reply:
and don't worry, this code only executes when there are two ZKs and we don't have two ZKs in corp or prod.